docker/entrypoint_check.sh

```diff
@@ -1,22 +1,32 @@
 #!/bin/sh -l
 
 echo '\nGetting the code...\n'
-git clone --quiet https://$INPUT_TOKEN@github.com/$GITHUB_REPOSITORY check
+git clone --quiet https://$PAT@github.com/$GITHUB_REPOSITORY check
 cd check
-git checkout $BRANCH_SOURCE
+git checkout $GITHUB_HEAD_REF
 rm .Rprofile
 
 echo '\nSession info\n'
 Rscript -e 'sessioninfo::session_info()'
 
+echo '\nCheck if only protocol is updated...\n'
+FOLDERNAME=$(Rscript -e 'protocolhelper::get_path_to_protocol("'$GITHUB_HEAD_REF'")')
+FOLDERNAME_REL=${FOLDERNAME#/home/runner/work/protocolsource/}
+echo 'FOLDERNAME=' $FOLDERNAME
+echo 'FOLDERNAME_REL=' $FOLDERNAME_REL
+CHANGED=$(git diff --name-only main | grep -v NEWS\.md | grep -v \.zenodo\.json| grep -v ^$FOLDERNAME_REL/)
+
+if [ -n "${CHANGED}" ]; then
+  echo '\nBranch '$GITHUB_HEAD_REF' contains changes to files which do not belong to protocol '$GITHUB_HEAD_REF', and these changes will not be passed to the website. Please remove these changes from this branch and start a new branch for changes to other protocols. Changes to be removed:' $CHANGED
+  exit 1
```

Comment:
Can you double check the use of single quotation marks? Are nested single quotation marks allowed?